superset-frontend/spec/javascripts/components/TableSelector_spec.jsx

```diff
@@ -18,267 +18,275 @@
 
-    afterEach(() => {
-      wrapper.instance().fetchTables.restore();
+  describe('change table', () => {
+    beforeEach(async () => {
+      fetchMock.get(GET_TABLE_NAMES_ENDPOINT, tables, {
+        overwriteRoutes: true,
+      });
+      act(() => {
+        wrapper
+          .find('[name="select-schema"]')
+          .first()
+          .props()
+          .onChange(selectedSchema);
+      });
+      await waitForComponentToPaint(wrapper);
+      wrapper.update();
     });
 
-    it('test 1', () => {
-      wrapper.instance().changeTable({
-        value: 'birth_names',
-        schema: 'main',
-        label: 'birth_names',
-        title: 'birth_names',
+    it('should change table value', async () => {
+      act(() => {
+        wrapper
+          .find('[name="select-table"]')
+          .first()
+          .props()
+          .onChange(selectedTable);
       });
-      expect(wrapper.state().tableName).toBe('birth_names');
+      await waitForComponentToPaint(wrapper);
+      wrapper.update();
+      expect(
+        wrapper.find('[name="select-table"]').first().props().value[0],
+      ).toEqual(selectedTable);
     });
 
-    it('should call onTableChange with schema from table object', () => {
-      wrapper.setProps({ schema: null });
-      wrapper.instance().changeTable({
-        value: 'my_table',
-        schema: 'other_schema',
-        label: 'other_schema.my_table',
-        title: 'other_schema.my_table',
+    it('should call onTableChange with schema from table object', async () => {
+      act(() => {
+        wrapper
+          .find('[name="select-table"]')
+          .first()
+          .props()
+          .onChange(selectedTable);
       });
-      expect(mockedProps.onTableChange.getCall(0).args[0]).toBe('my_table');
-      expect(mockedProps.onTableChange.getCall(0).args[1]).toBe('other_schema');
+      await waitForComponentToPaint(wrapper);
+      wrapper.update();
+      expect(mockedProps.onTableChange.getCall(0).args[0]).toBe('birth_names');
+      expect(mockedProps.onTableChange.getCall(0).args[1]).toBe('main');
     });
   });
 
-  it('changeSchema', () => {
-    sinon.stub(wrapper.instance(), 'fetchTables');
+  describe('getTableNamesBySubStr', () => {
+    beforeEach(() => {
+      wrapper.setProps({ schema: 'main' });
+    });
+    afterEach(fetchMock.resetHistory);
+    afterAll(fetchMock.reset);
 
-    wrapper.instance().changeSchema({ label: 'main', value: 'main' });
-    expect(wrapper.instance().fetchTables.callCount).toBe(1);
-    expect(mockedProps.onChange.callCount).toBe(1);
-    wrapper.instance().changeSchema();
-    expect(wrapper.instance().fetchTables.callCount).toBe(2);
-    expect(mockedProps.onChange.callCount).toBe(2);
+    it('should handle empty', () =>
+      act(() => {
+        wrapper
+          .find('[name="async-select-table"]')
+          .first()
+          .props()
+          .loadOptions()
+          .then(data => {
+            expect(data).toEqual({ options: [] });
```

Comment:
since this is part of a promise, I'm not sure if the test is exiting before it gets to this line. Looking at [the docs](https://jestjs.io/docs/en/asynchronous) it appears we can return a promise and jest will wait for it to finish before finishing the test. Since this is wrapped in an `act`, I'm not sure how to do that. We may need to use `done`, or make this an async function and use a few awaits, or setup the test with `expect.assertions(1)`.